chef/cookbooks/neutron/recipes/common_config.rb

```diff
@@ -185,56 +180,61 @@
 num_vlans = neutron[:neutron][:num_vlans]
 vlan_end = [vlan_start + num_vlans - 1, 4094].min
 
+
 case neutron[:neutron][:networking_plugin]
-when "openvswitch", "cisco"
-  agent_config_path = "/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini"
+when 'ml2'
+  ml2_mech_drivers = neutron[:neutron][:ml2_mechanism_drivers]
+  case
+  when ml2_mech_drivers.include?("openvswitch")
+    agent_config_path = "/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini"
 
-  directory "/etc/neutron/plugins/openvswitch/" do
-     mode 00755
-     owner "root"
-     group neutron[:neutron][:platform][:group]
-     action :create
-     recursive true
-     not_if { node[:platform] == "suse" }
-  end
+    directory "/etc/neutron/plugins/openvswitch/" do
+      mode 00755
+      owner "root"
+      group neutron[:neutron][:platform][:group]
+      action :create
+      recursive true
+      not_if { node[:platform] == "suse" }
+    end
 
-  template agent_config_path do
-    cookbook "neutron"
-    source "ovs_neutron_plugin.ini.erb"
-    owner "root"
-    group neutron[:neutron][:platform][:group]
-    mode "0640"
-    variables(
-      :physnet => neutron[:neutron][:networking_mode] == 'gre' ? nil : "br-fixed",
-      :networking_mode => neutron[:neutron][:networking_mode],
-      :vlan_start => vlan_start,
-      :vlan_end => vlan_end
+    template agent_config_path do
+      cookbook "neutron"
+      source "ovs_neutron_plugin.ini.erb"
+      owner "root"
+      group neutron[:neutron][:platform][:group]
+      mode "0640"
+      variables(
+        :physnet => neutron[:neutron][:ml2_type_drivers].include?('gre') ? nil : "br-fixed",
+        :networking_mode => neutron[:neutron][:networking_mode],
+        :vlan_start => vlan_start,
+        :vlan_end => vlan_end
```

Comment:
I would like to fixed the cleanup things when we are done with this PR. The change is already huge...